[Exporter] create folder if it doesnt exists instead of raising an error

techfreaque opened the pull request.

GuillaumeDSM commented:
Once this test is passing, we can merge it !
https://github.com/Drakkar-Software/OctoBot-Tentacles-Manager/runs/8235204721?check_suite_focus=true#step:5:173

techfreaque commented:
> Once this test is passing, we can merge it ! https://github.com/Drakkar-Software/OctoBot-Tentacles-Manager/runs/8235204721?check_suite_focus=true#step:5:173

Looks like its an issue happening only on linux, here on my windows its passing all tests.
Do you know how to fix it? If not I'll set up linux enviroment to debug it.
And also I fixed some tests that were not working in the last commit

Herklos commented on octobot_tentacles_manager/api/inspector.py:
```diff
@@ -26,6 +26,26 @@ def get_installed_tentacles_modules() -> set:
     return set(tentacle for tentacle in loaders.get_tentacle_classes().values())
 
 
+def get_installed_tentacles_modules_dict() -> dict:
```
What will it be used for?

techfreaque replied:
this can be used with the react ui to manage installed tentacles
meanwhile i moved it to the react ui

I also updated the pull request - I removed create folder if it doesnt exist as its already fixed in master

Herklos replied:
Why can't you use `get_installed_tentacles_modules` to do it?

techfreaque replied:
its returning as a set and tentacle_type is an object which doesnt work with json without changing it

Herklos replied:
Why not converting to json in client (caller) side?

Herklos commented on octobot_tentacles_manager/api/configurator.py:
```diff
@@ -161,3 +161,10 @@ def get_code_hash(tentacles: list) -> str:
     return util.get_tentacles_config_hash(tentacles, tentacles_setup_config)
+
+
+def get_installation_context_octobot_version() -> dict:
+    setup_config = configuration.TentaclesSetupConfiguration()
+    available_tentacle = util.load_tentacle_with_metadata(constants.TENTACLES_PATH)
+    setup_config.fill_tentacle_config(available_tentacle, constants.TENTACLE_CONFIG_FILE_NAME)
+    return setup_config.installation_context
```
👍

Herklos commented on requirements.txt:
```diff
@@ -11,7 +11,7 @@ aiofiles>=0.6.0, <0.7
 
 # Templating requirements
-jinja2>=3.0.3
+jinja2>=3.1.2
```
Why did you bumped jinja2?

techfreaque commented:
regarding your questions:

- [Tests] fix unknow octobot version tests
        why a new api ? if it's just for tests, we should create a test method
1. I moved it to tests as its only required there

- [Api] get installed tentacles as dict
        why get_installed_tentacles_modules_dict ? (keys should be constants)
1. I want to get the data ready to convert to json without needing to remove/modify objects in it. And I replaced it with an enum
